dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinUserMetadataRestController.java

```diff
@@ -303,19 +308,87 @@ private void sendEmailWithDownloadLink(Context context, DSpaceObject dso,
+        // notified, if the mail is not really sent (if it fails HERE, not later, e.g. due to mail server issue).
+        sendAdminNotificationEmail(context, downloadLink, dso, clarinLicense, mailType, clarinUserMetadataRestList);
+
+    }
+
+    private void sendAdminNotificationEmail(Context context,
+                                            String downloadLink,
+                                            DSpaceObject dso,
+                                            ClarinLicense clarinLicense,
+                                            MailType mailType,
+                                            List<ClarinUserMetadataRest> extraMetadata)
+            throws MessagingException, IOException {
+        try {
+            Locale locale = context.getCurrentLocale();
+            Email email2Admin = Email.getEmail(I18nUtil.getEmailFilename(locale, "clarin_download_link_admin"));
+            String ccAdmin = configurationService.getProperty("download.email.cc");
+            List<String> ccEmails = new ArrayList<>();
+            if (ccAdmin != null && !ccAdmin.isEmpty()) {
+                ccEmails.add(ccAdmin);
+            }
+            String licenseName = clarinLicense.getName().trim().replace(" ", "_").toLowerCase();
+            String[] licenseSpecialRecipients = configurationService.getArrayProperty("download.email.cc."
+                    + licenseName);
+            if (licenseSpecialRecipients != null) {
+                for (String cc : licenseSpecialRecipients) {
+                    if (!cc.isEmpty()) {
+                        ccEmails.add(cc.trim());
+                    }
+                }
+            }
+
+            if (!ccEmails.isEmpty()) {
+                for (String cc : ccEmails) {
+                    email2Admin.addRecipient(cc);
+                }
+                if (mailType == MailType.ALLZIP) {
+                    email2Admin.addArgument("all files requested");
+                } else if (mailType == MailType.BITSTREAM) {
+                    email2Admin.addArgument(dso.getName());
+                } else {
+                    throw new IllegalArgumentException("The mail type is not supported.");
+                }
+                email2Admin.addArgument(downloadLink);
+                email2Admin.addArgument(clarinLicense.getDefinition());
+                if (context.getCurrentUser() != null) {
+                    email2Admin.addArgument(context.getCurrentUser().getFullName());
+                    email2Admin.addArgument(context.getCurrentUser().getEmail());
+
+                } else {
+                    email2Admin.addArgument("Anonymous user");
+                    email2Admin.addArgument("Anonymous user");
+                }
+                StringBuilder exdata = new StringBuilder();
+                for (ClarinUserMetadataRest data : extraMetadata) {
+                    exdata.append(data.getMetadataKey()).append(": ").append(data.getMetadataValue()).append(", ");
```

Comment:
would String format be more readable?

Reply:
In my opinion not significantly. It would be, if we had fstring as in python or C# (f"{data.getMetadataKey()} : {data.getMetadataValue()}"),
but since we don't have that, one still must think about it. 
String builder could be faster, but it is insignificant in this number of appends.
I think it does not matter. I can use this:

exdata.append( String.format("%s : %s", data.getMetadataKey(), data.getMetadataValue());

if you want, but I do not think it really helps anything.